Add loc to admin panel

kylebechtel77 opened the pull request.

btidwell commented:
Will leave this script out of the pencilblue fork to avoid including it in our multi tenancy commit back to pencilblue/master

brianhyder committed:
0.5.0 Cleanup

brianhyder committed:
Merge branch 'master' into cleanup

brianhyder committed:
Finished pb.users calls.  Start on ls.get

brianhyder committed:
Unit Test fixes

brianhyder committed:
Merge branch 'master' into cleanup

Conflicts:
	include/service/entities/user_service.js

Commit:
initial commit. adds active theme and site aware locs and an admin panel to manage localization files

Commit:
missed a route in the admin panel

Commit:
missed route, fixed formatting of loc file

Commit:
fixing ls.g not registering on /about page in sample

Commit:
missed moving over a file

Commit:
updating names and error handling in file writter

Commit:
adding get route to get localizations for the front end. started adding db stuff

Commit:
fixing file path join issue

Commit:
Revamped the front end of the Localization page.  Removed a lot of the redirect logic that is no longer needed.  More cutting of junk is needed.

Commit:
updated localization to take in multiple localization kvs

Commit:
updates to retrieve and add things to the db

Commit:
Localization now is on the admin nav, the add view is done minus pulling from db, started on viewing view.

Commit:
passing site into getLocalization

Commit:
changing route to use qsvars

Commit:
forgot a !

Commit:
missed a var

Commit:
fixing getlocalizations and moving dao call in localization

Commit:
Changed routes for get locales to return josn to the front end.  Updated much of the html and angular to accept and respond to the get requests

Commit:
Changes... I dont remember what these where exactly...

Commit:
The front end view now pulls correctly in, all of the existing site overrides for the plugin and will update based on plugin and language.  More cleaning up of the angular code to follow.  Also, the 'view' page is framed, but not functional (copied from the site form)

Commit:
broken db changes

Commit:
fix id for save thingys

Commit:
finalizing view and fixing minor bugs. created and implemented view page

Commit:
Major refactor of the get and localization services.  Will rename after PR.  This makes the Database solution work.  The Post and Get Route now can support db storage.

Commit:
Merge remote-tracking branch 'origin/dbcrap' into dbcrap

# Conflicts:
#	plugins/pencilblue/controllers/actions/admin/sites/getLocalization.js
#	plugins/pencilblue/controllers/actions/admin/sites/localization.js

Commit:
Prevented the file path to be called after the DB path

kylebechtel77 committed:
Merge pull request #1 from kylebechtel77/dbcrap

Dbcrap

Commit:
Minor fix to allow for the front end to load the site specific locales from the db.  Updated the way the object is handled in angular.

Commit:
error handled if no db collection is found

Commit:
passing in post again, missed that when moving to a function

Commit:
added in dashes to locale and language for the object creation

Commit:
weird issues with the db in localization

Commit:
moving db call to base controller temporarily.

kylebechtel77 committed:
Merge pull request #2 from kylebechtel77/locale-db

Locale db

Commit:
more key changes

Commit:
Fixing keys

Commit:
Added in a missing if check.

Commit:
fixed another is key issue

Commit:
Added in locale keys for the tabs in the localization view, and fixed the update of objects.  They no longer just replace, they pull what is in the DB, update, and then replace in the db.

brianhyder committed:
Merge branch 'master' into cleanup

brianhyder committed:
More localization conversion

brianhyder committed:
Localization cleanup

brianhyder committed:
Style cleanup and more localization cleanup

brianhyder committed:
Fixed broken tests

Commit:
merged getlocalization and localization files. Still more refactoring needed

brianhyder committed:
More cleanup along with settings api

brianhyder committed:
POST and PUT but need to get delete working

Commit:
changing everything to use site instead of sitename

brianhyder committed:
Completed Setting Service api

brianhyder committed:
Documentation cleanup

brianhyder committed:
Merge pull request #939 from pencilblue/cleanup

Cleanup

Commit:
some refactoring of the localizatoin file

kylebechtel77 committed:
Merge pull request #3 from kylebechtel77/clean-up

Clean up

wesanderson303 committed:
Init'ed plugins detect setting discrepancies and add them accordingly

wesanderson303 committed:
autoSync defaults to off

wesanderson303 committed:
Removed settings are handled and updated on spinup

brianhyder committed:
AdminNavigationApi - Provided Get functionality

brianhyder committed:
Merge branch 'master' into 0.5.1

brianhyder committed:
Merge branch '0.5.1' into feature/AdminNavigationApi

Conflicts:
	include/service/base_object_service.js

brianhyder committed:
Merge pull request #943 from pencilblue/feature/AdminNavigationApi

Feature/admin navigation api

wesanderson303 committed:
Slightly optimizing the discrepancy detection

wesanderson303 committed:
PluginService.syncSettings

wesanderson303 committed:
A little cleaner code

wesanderson303 committed:
Updates for efficiency and readability

wesanderson303 committed:
oops - removing 'autoSync' setting

Commit:
fixed file create bug

Commit:
Added in the ability to have nav localize as well as the page (content page) to localize.  They are all setting wrapped from the PB config.  I can look for deeper solutions if these are not adequate.

Commit:
Added a flag to turn on reprocess by default so that we dont need to manually set it in each and every file/controller but still keeping the other flags to allow for it to be used at a 'piece by piece' level

Commit:
Updating the changes to the config file

Commit:
updating configuration options

Commit:
Fixed the drop downs on the localization page to show the default selected option.  So the default locale will default in the language box, and the active theme will default into the select plugin box.
Next step is the clean up the html for this page

Commit:
unneeded flag in writeFile

Commit:
Added localizations to all of our templates for the localization view.  Moved the templates to their own folder and not site.  Will revamp and clean the angular and controller next

Commit:
Merge pull request #4 from kylebechtel77/clean-up

Tested locally, and it works.  The french file was created for me and the server did not go down.

Commit:
Removing the wrong config file

Commit:
adding base template service reprocessing config

kylebechtel77 committed:
Merge pull request #5 from kylebechtel77/localize-nav-and-content

Localize nav and content. Looks good. Thanks for adding the flags

Commit:
Merge remote-tracking branch 'origin/hackathon' into admin-cleanup

kylebechtel77 committed:
Merge pull request #6 from kylebechtel77/clean-up

adding base template service reprocessing config

Commit:
Moved the angular file to the correct folder instead of being under site to match the html files

Commit:
Removed the global set reprocess true flag and logic as turning it on breaks a number of the content pages.

Commit:
Spacing, literally.  Added a space between the flopy icon and the word save on the save button

Commit:
Merge branch 'hackathon' of github.com:kylebechtel77/pencilblue into admin-html-cleanup

Commit:
removing modal, merging with master, and adding proper configs

kylebechtel77 committed:
Merge pull request #7 from kylebechtel77/admin-html-cleanup

Admin html cleanup

kylebechtel77 committed:
Update en-us.json

kylebechtel77 committed:
Update en-US.json

Commit:
removing unneeded testing files

Commit:
fixing merge conflicts in .json files

brianhyder committed:
Merge pull request #946 from careerbuilder/autosync-plugin-settings

Autosync plugin settings

brianhyder committed:
Merge branch 'master' into 0.5.1

Commit:
updating routes, flags, and putting localization service in a different localition in request handler so it is initialized with site.

Commit:
code climate fixes

Commit:
moved localization into localization api controller and broke it down into smaller services.

Commit:
fixed route and require statement

Commit:
polishing the new db and file localization services.

kylebechtel77 committed:
Merge pull request #8 from kylebechtel77/breakdown-loc-api

Breakdown loc api

brianhyder committed:
Merge branch 'master' into 0.5.1

brianhyder committed:
#958 - Corrects issue where created date is saved as string on media edit

brianhyder committed:
Merge pull request #961 from pencilblue/issue/958

Fixes #958 - Corrects issue where created date is saved as string on media …

Commit:
initial setup of job runner

brianhyder committed:
#962 Cleans up MemoryEntityService to prevent single instance leaks.  Still needs to correct issue for change handler leaks

brianhyder committed:
#962 - Corrects bug that prevents caching of plugin settings

brianhyder committed:
#962 - Fixes more places that incorrectly reference cache timeouts

brianhyder committed:
#962 - Updates from code review and code climate

blakecallens committed:
fixes #955 - generated URLs are safe

brianhyder committed:
#962 - Added process dependency back because 0.12 needs it

brianhyder committed:
Merge pull request #964 from pencilblue/issue/962

Fixes #962 - Corrects Memory Leak & Performance Issues Around Analytics

blakecallens committed:
Merge pull request #965 from pencilblue/issue955

fixes #955 - generated URLs are safe

n1ru4l committed:
Adds comment content validation

Comments could be saved into the database with empty content.
Now the new_comment api controller will check for a min-length of 5 characters.

Commit:
This changes the behavior of redirects from simply showing you the same site (even though your url is different) to doing a true redirect.

Commit:
updated to use status codes package

brianhyder committed:
Merge pull request #967 from kylebechtel77/redirects

This changes the behavior of redirects

brianhyder committed:
Updated versions to for initial branch to create version

brianhyder committed:
Merge pull request #966 from n1ru4l/patch-1

Adds comment content validation

brianhyder committed:
Updating for merge to master

brianhyder committed:
Cleanup from code climate and documentation warning fixes

brianhyder committed:
Final cleanup item before merge

brianhyder committed:
Merge pull request #973 from pencilblue/0.5.1

0.5.1

brianhyder committed:
Merge branch 'master' into 0.6.0

Conflicts:
	package.json
	yuidoc.json

brianhyder committed:
Hot-fix to correct double registration of GA plugin

brianhyder committed:
#981 - Adds DAO support for injecting in an "id" field.

Commit:
beginning to prune job runner

Commit:
changing db call to the localization init function in preperation for the job runner.
changing the merge function so it actually removes things on save. adding a .g to sample

Commit:
Merge branch 'staging' of github.com:kylebechtel77/pencilblue into admin-cleanup

# Conflicts:
#	include/config.js
#	include/service/entities/template_service.js
#	plugins/pencilblue/templates/angular/admin/localization/localization_settings.html

Commit:
The database now supports deletions.  At the start of save, it takes the object from the database and the payload from the post.  It will then iterate through each key in the DB object.  If it finds that the key has the language/country values, it will delete the country as a whole.

(we can expand this to respect plugin if we want to.)

It will recursively (sorta) delete the empty parent structure, so if your key only has en-US translations, it will delete first the US object, then checks if the en is empty, which it would be, then that Object is deleted, and then the key is checked, if it only has one property (isKey) it will be deleted as well.

This last bit was to provide for cleanup to make sure all remains of deleted localizations are removed from the object structure.

The normal flow is then respected, at which point every object that was submitted to the backend will be created and shoved onto the DB object and then written back to the DB.

Commit:
Merge branch 'staging' of github.com:kylebechtel77/pencilblue into init-delete

Commit:
Added support for the plugin level of the localization settings.  Did some minor cleanup to the function removeAllLocalesFromDoc

Commit:
Moved the function, added additional comments

brianhyder committed:
#981 - Adds the ability to rely upon the "id" variable instead of the _id property

brianhyder committed:
#982 - Adds topic filter to article and page APIs

brianhyder committed:
#982 - started media api controller and added some documentation

Commit:
pulling job runner all the way through for the db.

Commit:
fixed the database query in the job runner

Commit:
Broke out the DB calls in the init function into a separate function that is called at the end of the file call only if the db flags are turned on.  Passes through the callback and the compounded results, otherwise it will call the callback as normal.

kylebechtel77 committed:
Merge pull request #10 from kylebechtel77/init-delete

Init delete

Commit:
merging with staging

Commit:
Merge pull request #11 from kylebechtel77/loc-job-runner

Loc job runner

Commit:
merged with pb

kylebechtel77 committed:
Merge pull request #12 from kylebechtel77/merge

Merge

Commit:
Added back in the Job Runner's modal for the front end.  It does not finish, there are still bugs to be worked out.  It appears that the job runner is not calling its 'done' commands correctly.

Commit:
finishing job runner register command function

Commit:
updating some text and removing console.logs

kylebechtel77 committed:
Merge pull request #13 from kylebechtel77/modal-fix

Modal fix

Commit:
TODO changed to DONE

Commit:
fixed fun issue where site delete fails if site doesnt already have a record.

kylebechtel77 committed:
Merge pull request #14 from kylebechtel77/testing

Testing

kylebechtel77 committed:
Merge pull request #9 from kylebechtel77/staging

Staging

Commit:
Found obscure bug where site ID was not an object being passed to set site.  This caused the site to be set to undefined, and thus caused an issue for the updateLocalesCommand where an undefined record (what should have been the site) would be added to storage on completion of the job runner.  This caused massive errors when trying to localize text on the page (when the job runner finished, it would refresh the page, and 500)

Commit:
Merge branch 'add-loc-to-admin-panel' of github.com:kylebechtel77/pencilblue into testing

brianhyder committed:
#982 - Added media endpoints implementation

brianhyder committed:
Fixes #985 - corrects issue that prevents article api from working

brianhyder committed:
Merge pull request #987 from pencilblue/issue/985

Fixes #985 - corrects issue that prevents article api from working

brianhyder committed:
Merge pull request #983 from pencilblue/feature/981

Fixes #981

brianhyder committed:
Merge branch 'master' into 0.6.0

brianhyder committed:
Merge pull request #986 from pencilblue/feature/982

Fixes #982 - Adds topic filter for Article, Page, Media

Commit:
There is a memory enhancement in localization that keeps track of all the keys in a quick look up.  Localization.keys  This object is filled when the files are loaded in the init function.  toplayer.key is the structure.  Our database solution did not update records in this object structure (localization.keys).  This adds code to include our records into this object structure. site.key and just key to make sure that we are compatible with both .g and .get and for locales that are written in the form ^loc_my_key^ and ^loc_site.my_key^

brianhyder committed:
Fixes #989 - Adds the about me endpoint

brianhyder committed:
Merge pull request #991 from pencilblue/feature/989

Fixes #989 - Adds the about me endpoint

Commit:
fixing code climate issues

kylebechtel77 committed:
Merge pull request #15 from kylebechtel77/testing

fixing code climate issues

Commit:
adding sinon for stubbing and updating localization tests

kylebechtel77 committed:
Merge pull request #16 from kylebechtel77/testing

adding sinon for stubbing and updating localization tests

Commit:
Merge branch 'add-loc-to-admin-panel' of github.com:kylebechtel77/pencilblue into testing

Commit:
Merge pull request #17 from kylebechtel77/key-checks

Key checks

Commit:
Merge branch '0.6.0' of https://github.com/pencilblue/pencilblue into add-loc-to-admin-panel

Commit:
merging with 0.6.0

Commit:
updating package.json

Commit:
updating details.json so we can actually install plugins

Commit:
Fixed "generic" keys list in the admin panel.  It now will show the default key/values from the files that are loaded.

Commit:
Merge pull request #18 from kylebechtel77/generic-key-admin-panel-fix

Fix: Admin Panel View
